picard/script/functions.py

```diff
@@ -286,6 +286,38 @@ def func_replace(parser, text, old, new):
     return text.replace(old, new)
 
 
+@script_function(documentation=N_(
+    """`$replacemulti(name,search,replace,separator="; ")`
+
+Replaces occurrences of `search` with `replace` in the multi-value tag `name`.
+
+Example:
+
+    $replacemulti(genre,Idm,IDM)
+"""
+))
+def func_replacemulti(parser, name, search, replace, separator=MULTI_VALUED_JOINER):
+    if not name or not separator:
+        return ""
+
+    name = normalize_tagname(name)
+    multi = parser.context.get(name)
+    if not multi:
+        return ""
+
+    multi = multi.split(separator)
+    changed = False
+    for n, value in enumerate(multi):
+        if value == search:
+            multi[n] = replace
+            changed = True
+
+    if changed:
+        parser.context[name] = separator.join(multi)
```

Comment:
If I read this code correctly, this actually changes the variable provided.

Would it not be better to have a usage like `$setmulti(genre,$replacemulti(%genre%,Idm,IDM))`?

Reply:
Yes, referencing the variable by name was how I intended the function to be used, as that made the most sense to me. I did consider what you have suggested, but I didn't go with it because a) it's more complex/verbose to use and less readable (IMO) and b) I can't really think of a use case where someone would take the value of an existing multi-value variable, replace an element within it and then assign it to a different variable.

I'm not wedded to this approach so if there is a convention I'm missing then I am happy to change it.

Reply:
Yes - I think that there is a convention.

And although your approach is less verbose, it is a different approach to most functions which return the value. And it is only less verbose if you want to change the existing variable - it is more verbose and even less conventional if you want to assign the changed contents to a different variable:
```
$setmulti(new_genre,$replacemulti(%genre%,Idm,IDM))
vs.
$setmulti(new_genre,%genre%)
$replacemulti(new_genre,Idm,IDM)
```

Reply:
Thanks for the comments, I believe the implementation should now work as per your approach.